src/pages/view-test-result/view-test-result.ts

```diff
@@ -1,26 +1,122 @@
-import { Component } from '@angular/core';
-import { IonicPage, NavController, NavParams, Platform } from 'ionic-angular';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { IonicPage, NavController, NavParams, Platform, Loading, LoadingController } from 'ionic-angular';
 import { BasePageComponent } from '../../shared/classes/base-page';
 import { AuthenticationProvider } from '../../providers/authentication/authentication';
+import { SearchProvider } from '../../providers/search/search';
+import { StandardCarTestCATBSchema, ApplicationReference } from '@dvsa/mes-test-schema/categories/B';
+import { tap, catchError } from 'rxjs/operators';
+import { of } from 'rxjs/observable/of';
+import { TestDetailsModel } from './components/test-details/test-details.model';
+import { inflateSync } from 'zlib';
+import { Subscription } from 'rxjs/Subscription';
+import { DateTime } from '../../shared/helpers/date-time';
+import { ExaminerDetailsModel } from './components/examiner-details/examiner-details.model';
 
 @IonicPage()
 @Component({
   selector: 'page-view-test-result',
   templateUrl: 'view-test-result.html',
 })
-export class ViewTestResultPage extends BasePageComponent {
+export class ViewTestResultPage extends BasePageComponent implements OnInit, OnDestroy {
 
   applicationReference: string = '';
 
+  testResult: StandardCarTestCATBSchema;
+  testDetails: TestDetailsModel;
+  examinerDetails: ExaminerDetailsModel;
+
+  isLoading: boolean;
+  loadingSpinner: Loading;
+  subscription: Subscription;
+
   constructor(
     public navController: NavController,
     public platform: Platform,
     public navParams: NavParams,
+    public loadingController: LoadingController,
     public authenticationProvider: AuthenticationProvider,
+    public searchProvider: SearchProvider,
+
   ) {
     super(platform, navController, authenticationProvider);
 
     this.applicationReference = navParams.get('applicationReference');
   }
 
+  ngOnInit(): void {
+    this.handleLoadingUI(true);
+
+    this.subscription = this.searchProvider
+      .getTestResult(this.applicationReference, this.authenticationProvider.getEmployeeId())
+      .pipe(
+        tap((data) => {
+          const buffer = Buffer.from(data, 'base64');
+          const decompressedData = inflateSync(buffer).toString();
+          this.testResult = JSON.parse(decompressedData) as StandardCarTestCATBSchema;
```

Comment:
Perhaps this extra tap phase could be hidden behind another provider which will simply give you the `StandacrdCarTestCATBSchema` object?